File: packages/sdk-java/src/main/java/com/cdpus/sdk/Transport.java
package com.cdpus.sdk;

import java.io.IOException;

public interface Transport {
  /**
   * Sends a JSON request body to the CDP endpoint.
   *
   * @param url full /v1/track URL
   * @param jsonBody serialized batch
   * @return HTTP status code
   */
  int send(String url, String jsonBody) throws IOException, InterruptedException;
}

File: packages/sdk-php/src/Transport.php
<?php

declare(strict_types=1);

namespace CdpUs;

interface Transport
{
    /**
     * Sends a JSON body to the CDP endpoint.
     *
     * @example
     * $status = $transport->send('https://api.example.com/v1/track', '{"events":[]}');
     */
    public function send(string $url, string $jsonBody): int;
}
